### ISSUE TITLE
Uninstalling Wfp Tool Does Not Return System to the State Prior to Installation of Wfp Tool

### ISSUE THREAD

**Issue by HJLBX** [henrypp/simplewall#3]
WARNING !!

Uninstalling Wfp Tool does not return Windows Firewall and filtering for programs to a state prior to installing Wfp Tool unless...

When you uninstall Wfp Toll - BEFORE DOING SO - you must:
1. untick "Disable Windows FIrewall"; and
2. Uninstall filtering

Otherwise two things will happen:

A. Windows Firewall and associated services will remain disabled and cannot be started (Error 0x8007042c) https://support.microsoft.com/en-us/kb/2530126
B. Programs for which there was not an Allow rule created with Wfp Tool will not be able to connect to the internet

In other words, if you do not do the above, the disabled Windows Firewall and program filtering will persist on the system (in BFE Service\registry ?) AFTER uninstalling Wfp Tool.

You can correct this by reinstalling Wfp Tool and doing 1 and 2 above, but for 2 you must recreate all the rules that were present at the time you initially uninstalled it and then uninstall the filtering.

This issue is a real problem.

**Comment by henrypp** [comment]
wfptool will disable "mpssvc" AND "mpsdrv" services.
after you push "uninstall" button you can check "enable window firewall" checkbox.

**Comment by henrypp** [comment]
try to run debug view before uninstalling (https://technet.microsoft.com/en-us/sysinternals/debugview.aspx)
if you push uninstall button and checkbox is checked then you click YES and no "failed" messages in debug view is output, then problem on you.

**Comment by HJLBX** [comment]
Here is MEGA download link of video:  [https://mega.nz/#!xVpQDZ4J!q-lEGT5dJEGK4_Iph10e87xRwLot82lKqroUZBYEHZ4](url)

I can reproduce the issue shown in video _at will_ every time.

I can uninstall all other softs and reproduce the issue shown in video _at _will_ every time.

**Comment by HJLBX** [comment]
LOL... for whatever reason GitHub does not open MEGA link, but instead re-directs to 404.

Not making it easy for us...

To access MEGA:

1. Highlight URL with cursor (blue color)
2. Right-click
3. Copy
4. Open new tab in browser
5. Paste URL

**Comment by henrypp** [comment]
fairly simple thing you describe, just a copy url in address bar paste them and click enter is all it is, not nearly as complicated as, say, for instance, highlight url with cursor.

for whatever reason my provider blocking mega.nz thats the reason why i'm cannot open the link.

ps: it's like the safety lecture, one of my favorite part of the plane ride, especially that part where they teach us how to use the seat belt, they are actually taking time out to describe the intricate workings of a belt buckle.

**Comment by HJLBX** [comment]
Numbered procedure is a habit of mine.  I understand "seat belt."

Can you access ZippyShare ?

If yes, I will create download links.

**Comment by henrypp** [comment]
@HJLBX yes

**Comment by HJLBX** [comment]
[http://www57.zippyshare.com/v/QaeAapPC/file.html](url)

**Comment by henrypp** [comment]
1) is a installer bug, because wfp tool will create own persistent provider to windows filtering platform and (un)installer cannot do this. salvation: push the "Uninstall filters" button in wfp tool window, so there you can restore windows firewall to its enabled state and remove persistent provider.
2) logging only for dropped packets. windows filtering platform will not for monitoring entire traffic.
3) continued ~~alerts~~ notifications happened because in 1.3.2 added inbound (from another computer) packets logging, i can solve this in next release by add new param to settings and power off this by default.
4) outbund icmp protocol (ping) and domain name service (dns) is allowed by default, you can switch this off in "settings" -> "system rules". if you uncheck dns you can ping only ip addressess.

**Comment by henrypp** [comment]
closed. see 1.5.5
